consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/headervalidationrules/ProofOfStakeRule.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Hyperledger Besu Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.consensus.merge.headervalidationrules;
+
+import org.hyperledger.besu.consensus.merge.MergeContext;
+import org.hyperledger.besu.ethereum.ProtocolContext;
+import org.hyperledger.besu.ethereum.core.BlockHeader;
+import org.hyperledger.besu.ethereum.core.Difficulty;
+import org.hyperledger.besu.ethereum.mainnet.AttachedBlockHeaderValidationRule;
+
+import java.util.Optional;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public abstract class ProofOfStakeRule implements AttachedBlockHeaderValidationRule {
```

Comment:
I don't think proof-of-stake should leak into the execution client.  We aren't actually doing proof of stake, we are deferring to an external consensus mechanism.  If besu were being used as an execution client in a rollup, there may or may not be proof-of-stake, it could just as easily be PoA etc

Reply:
fair point, easy fix, will adopt